jaas/src/main/java/com/griddynamics/jagger/jaas/rest/TestEnvironmentRestController.java

```diff
@@ -132,35 +125,23 @@ private void validateTestEnv(TestEnvironmentEntity testEnv) {
         if (!matcher.matches())
             throw new TestEnvironmentInvalidIdException(envId, envIdPattern);
 
-        if (testEnv.getRunningTestSuite() == null && testEnv.getStatus() == RUNNING
-                || testEnv.getRunningTestSuite() != null && testEnv.getStatus() == PENDING)
+        if (testEnv.getRunningTestSuite() == null && testEnv.getStatus() == TestEnvironmentEntity.TestEnvironmentStatus.RUNNING
+                || testEnv.getRunningTestSuite() != null && testEnv.getStatus() == TestEnvironmentEntity.TestEnvironmentStatus.PENDING)
             throw new WrongTestEnvironmentStatusException(testEnv.getStatus(), testEnv.getRunningTestSuite());
     }
 
     private void setExpiresHeader(HttpServletResponse response, TestEnvironmentEntity testEnv) {
-        response.addHeader("Environment-Expires", getFormattedExpirationDate(testEnv, HEADER_DATE_FORMAT));
-    }
-
-    private void setSessionCookie(HttpServletResponse response, TestEnvironmentEntity testEnv, int cookieMaxAgeSeconds) {
-        Cookie cookie = new Cookie(ENVIRONMENT_SESSION_COOKIE, testEnv.getSessionId());
-        cookie.setMaxAge(cookieMaxAgeSeconds);
-        cookie.setPath("/");
-        response.addCookie(cookie);
+        response.addHeader(TestEnvUtils.EXPIRES_HEADER, getFormattedExpirationDate(testEnv));
     }
 
-    private String getSessionCookie(HttpServletRequest request) {
-        Cookie[] cookies = request.getCookies();
-        if (cookies != null) {
-            for (Cookie cookie : cookies) {
-                if (cookie.getName().equals(ENVIRONMENT_SESSION_COOKIE)) {
-                    return cookie.getValue();
-                }
-            }
-        }
-        return null;
+    private Cookie getSessionCookie(TestEnvironmentEntity testEnv) {
+        Cookie cookie = new Cookie(TestEnvUtils.SESSION_COOKIE, testEnv.getSessionId());
+        cookie.setMaxAge(environmentsTtlMinutes * 60);
+    
```

Comment:
Expires time won't be updated without `cookie.setPath("/");`
Check it, please, and fix if I'm right.